packages/kilt-did/src/Did.utils.ts

```diff
@@ -0,0 +1,184 @@
+import { SDKErrors, Crypto } from '@kiltprotocol/utils'
+import type { Codec, Registry } from '@polkadot/types/types'
+import type {
+  DidSigned,
+  PublicKeyEnum,
+  ISigningKeyPair,
+  UrlEnum,
+  IPublicKey,
+  UrlEncoding,
+  IDidDeletionOptions,
+  IUpdateOptions,
+  IDidCreationOptions,
+  IAuthorizeCallOptions,
+} from './types'
+import type {
+  DidAuthorizedCallOperation,
+  DidCreationOperation,
+  DidDeletionOperation,
+  DidPublicKey,
+  DidUpdateOperation,
+} from './types.chain'
+
+export const KILT_DID_PREFIX = 'did:kilt:'
+
+export function getDidFromIdentifier(identifier: string): string {
+  return KILT_DID_PREFIX + identifier
+}
+
+export function getIdentifierFromDid(did: string): string {
+  if (!did.startsWith(KILT_DID_PREFIX)) {
+    throw SDKErrors.ERROR_INVALID_DID_PREFIX(did)
+  }
+  return did.substr(KILT_DID_PREFIX.length)
+}
+
+export function signCodec<PayloadType extends Codec>(
+  payload: PayloadType,
+  signer: ISigningKeyPair
+): DidSigned<PayloadType> {
+  const signature = {
+    [signer.type]: signer.sign(payload.toU8a()),
+  }
+  return { payload, signature }
+}
+
+export function formatPublicKey(keypair: IPublicKey): PublicKeyEnum {
+  const { type, publicKey } = keypair
+  return { [type]: publicKey }
+}
+
+export function isIKeyPair(keypair: unknown): keypair is IPublicKey {
+  if (typeof keypair === 'object') {
+    const { publicKey, type } = keypair as any
+    return publicKey instanceof Uint8Array && typeof type === 'string'
+  }
+  return false
+}
+
+export function encodeEndpointUrl(url: string): UrlEnum {
+  const typedUrl: Record<string, UrlEncoding> = {}
+  const matched = Array.from(['http', 'ftp', 'ipfs']).some((type) => {
```

Comment:
I guess here we might want to perform some more serious checks, like using existing regexes to parse those addresses.

Reply:
I disagree tbh. Reimplementing the string validation we're doing on the node feels like we'd be setting ourselves up for another maintenance nightmare where we risk that the node and sdk implementations are out of sync. And consumers of the sdk are incentivised to run their own checks by the transaction fees they waste every time they mess it up

Reply:
the checks that are run as part of this util are also only there because the url string needs to be transformed to something like this:
```
{
  http: {
    payload: 'https://example.com'
  }
}
``` 
Could we not just have `newEndpointUrl` be a string on `DidCreationOperation` and `DidUpdateOperation` ?